pkg/gateway/gatewaylwm2m/gatewaylwm2m.go

```diff
@@ -0,0 +1,241 @@
+package gatewaylwm2m
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+	"github.com/edgenesis/shifu/pkg/gateway/gatewaylwm2m/lwm2m"
+	"github.com/edgenesis/shifu/pkg/k8s/api/v1alpha1"
+	"github.com/edgenesis/shifu/pkg/logger"
+	"gopkg.in/yaml.v2"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/configmap"
+)
+
+const (
+	deviceShifuHost          = "http://localhost:8080"
+	ConfigmapFolderPath      = "/etc/edgedevice/config"
+	ConfigmapInstructionsStr = "instructions"
+	ObjectIdStr              = "ObjectId"
+	DataTypeStr              = "DataType"
+
+	pingIntervalSec = 10
+)
+
+type Gateway struct {
+	client     *lwm2m.Client
+	k8sClient  *rest.RESTClient
+	edgeDevice *v1alpha1.EdgeDevice
+}
+
+func New() (*Gateway, error) {
+	edgedevice, krclient, err := deviceshifubase.NewEdgeDevice(&deviceshifubase.EdgeDeviceConfig{
+		NameSpace:  os.Getenv("EDGEDEVICE_NAMESPACE"),
+		DeviceName: os.Getenv("EDGEDEVICE_NAME"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if edgedevice.Spec.GatewaySettings == nil {
+		return nil, fmt.Errorf("GatewaySettings not found in EdgeDevice spec")
+	}
+
+	client, err := lwm2m.NewClient(context.TODO(), lwm2m.Config{
+		EndpointName:    edgedevice.Spec.GatewaySettings.LwM2MSetting.EndpointName,
+		EndpointUrl:     *edgedevice.Spec.GatewaySettings.Address,
+		Settings:        *edgedevice.Spec.GatewaySettings.LwM2MSetting,
+		DeviceShifuHost: deviceShifuHost,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var gateway = &Gateway{
+		edgeDevice: edgedevice,
+		k8sClient:  krclient,
+		client:     client,
+	}
+
+	if err := gateway.LoadConfiguration(); err != nil {
+		return nil, err
+	}
+
+	return gateway, nil
+}
+
+// LoadCfg loads the configuration from the ConfigMap
+func (g *Gateway) LoadConfiguration() error {
+	// Load the configmap
+	cfg, err := configmap.Load(ConfigmapFolderPath)
+	if err != nil {
+		return err
+	}
+
+	var instructions deviceshifubase.DeviceShifuInstructions
+	if instructionInCfg, ok := cfg[ConfigmapInstructionsStr]; ok {
+		err := yaml.Unmarshal([]byte(instructionInCfg), &instructions)
+		if err != nil {
+			logger.Errorf("Error parsing %v from ConfigMap, error: %v", ConfigmapInstructionsStr, err)
+			return err
+		}
+	}
+
+	var objMap = make(map[string]*lwm2m.Object)
+	for instructionName, instruction := range instructions.Instructions {
+		// Skip if instruction is nil
+		if instruction == nil {
+			logger.Infof("Instruction %v is nil", instructionName)
+			continue
+		}
+
+		// Skip if instruction does not set the ObjectId
+		objectId, exists := instruction.DeviceShifuGatewayProperties[ObjectIdStr]
+		if !exists {
+			logger.Infof("Instruction %v does not have an ObjectId", instructionName)
```

Comment:
error info?

Reply:
ditto

Reply:
If error, change it into error, if just an normal info, I suggest just remove the log to avoid potential log flooding.